src/db/sysdb_search.c

```diff
@@ -975,7 +977,64 @@ int sysdb_getgrnam(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    if (sss_domain_is_mpg(domain)) {
+    switch (mpg_mode) {
```

Comment:
This whole logic has another alternative - only search for groups or users with no original GID in a single LDB search. But because the original GID attribute in sysdb is not indexed, I wasn't sure if it's worth the sysdb upgrade. Another talking point for the review..